vpr/src/route/rr_graph2.cpp

```diff
@@ -2400,10 +2220,10 @@ void load_sblock_pattern_lookup(const int i,
                     itrack = ichan % nodes_per_chan->x_list[j];
                 }
 
-                if ((*incoming_wire_label[side_cw])[itrack] != UN_SET) {
+                if ((*incoming_wire_label[side_cw])[itrack] != RRNodeId(UN_SET)) {
```

Comment:
This cast looks strange to me.  Is this data structure storing an rr_node index (RRNodeId) or an integer that represents something else?  If it is storing an RRNodeId, it seems like we should get rid of UN_SET and instead use RRNodeID::Invalid as the sentinel value for not set yet (and update the commenting to match).  

Right now both RRNodeId::Invalid() and UN_SET are -1 I think, so it will all work, but it seems strange to define a separate UN_SET invalid sentinel.

Reply:
If this data structure is indeed storing an RRNodeId it would be good to add a comment to that effect in this routine and in the data structure definition (unless there is one already, but I didn't find it).  From a quick look at the code I couldn't quite figure out if this incoming_wire_label data structure is storing some kind of track index, or if it's a unique rr_node id.  If you know the answer to that Xifan, it would be good to add a comment now.